### ISSUE TITLE
Hard-coded subject in mail helper

### ISSUE THREAD

**Issue by digime99** [sendgrid/sendgrid-csharp#234]
Please see sendgrid-csharp/SendGrid/SendGrid/Helpers/Mail/Mail.cs

```
  public Mail(Email from, String subject, Email to, Content content)
        {
            this.From = from;
            Personalization personalization = new Personalization();
            personalization.AddTo(to);
            this.AddPersonalization(personalization);
            **this.Subject = "Hello World from the SendGrid CSharp Library";**
            this.AddContent(content);
        }
```

Should be this...

`this.Subject = subject;`

**Comment by ncruces** [comment]
Not only is the API awful, but it is full of blunders such as this.

Who's idea was it to make a C# API full of:

- `dynamic`;
- `snake_cased` members, that violate every naming convention of the language;
- stuff not amenable to `async`/`await`;
- built on "helper classes" like `Mail` that make it impossible to use object/collection initializers.

How is this ***at all helpful*** as a C# API?

For now I'll just revert, but if this is the "future", I'll be wrapping the REST API directly through something like [Refit](https://github.com/paulcbetts/refit).

**Comment by thinkingserious** [comment]
@digime99 my apologies, I will be fixing that issue shortly. Thank you very much for letting us know!

@ncruces thanks for the specific feedback, it is much appreciated! We rewrote the library to support all v3 web api endpoints and the highly anticipated v3 /mail/send endpoint. This is a starting point and we are listening carefully and will update accordingly. After fixing the bug reported in this ticket, I'll create new tickets to address the other issues you mention. Hopefully we can then dig into the details and continue improvement.

**Comment by thinkingserious** [comment]
@digime99 this has been fixed in version 7.0.1: https://www.nuget.org/packages/SendGrid

**Comment by critot** [comment]
totally agree with ncruces, a full step back!
